mwdb/resources/remotes.py

```diff
@@ -0,0 +1,459 @@
+                self.on_created(item)
+            else:
+                hooks.on_reuploaded_object(item)
+                self.on_reuploaded(item)
+        finally:
+            item.release_after_upload()
+
+        logger.info(f'{self.ObjectType.__name__} added', extra={
+            'dhash': item.dhash,
+            'is_new': is_new
+        })
+        schema = self.ItemResponseSchema()
+        return schema.dump(item)
+
+
+class APiRemoteFilePullResource(APiRemotePullResource):
+    ObjectType = File
+    ItemResponseSchema = FileItemResponseSchema
+
+    on_created = hooks.on_created_file
+    on_reuploaded = hooks.on_reuploaded_file
+
+    @requires_authorization
+    def post(self, remote_name, identifier):
+        """
+        ---
+        summary: Pulls file from remote to local instance
+        description: |
+            Pulls file from the remote instance to the local instance
+        security:
+            - bearerAuth: []
+        tags:
+            - api_remote
+        parameters:
+            - in: path
+              name: remote_name
+              description: name of remote
+              schema:
+                type: string
+            - in: path
+              name: identifier
+              description: Object identifier (SHA256/SHA512/SHA1/MD5)
+              schema:
+                type: string
+        responses:
+            200:
+                description: Information about uploaded config
+                content:
+                  application/json:
+                    schema: FileItemResponseSchema
+            404:
+                description: When the search name of the remote instance is not figured in the application config
+            409:
+                description: Object exists yet but has different type
+        """
+        remote_url, api_key = get_remote_api_data(remote_name)
+        response = self.session.request("GET", f"{remote_url}/api/file/{identifier}",
+                                        headers={'Authorization': 'Bearer {}'.format(api_key)})
+        map_remote_api_error(response.status_code)
+        file_name = response.json()['file_name']
+
+        response = self.session.request("POST", f"{remote_url}/api/request/sample/{identifier}",
+                                        headers={'Authorization': 'Bearer {}'.format(api_key)})
+        map_remote_api_error(response.status_code)
+        download_url = response.json()['url']
+
+        response = self.session.request("GET", f"{remote_url}/api/{download_url}",
+                                        headers={'Authorization': 'Bearer {}'.format(api_key)},
+                                        stream=True)
+        map_remote_api_error(response.status_code)
+
+        with SpooledTemporaryFile() as file_stream:
+            # for chunk in response.iter_content(chunk_size=64):
+            #     file_stream.write(chunk)
+            file_stream.write(response.raw.read())
```

Comment:
That one should work as well.
```suggestion
            for chunk in response.iter_content(chunk_size=2**16):
                file_stream.write(chunk)
```